### PR TITLE
Try to eliminate dead stores

### PR DESCRIPTION
**Pull request by alshopov**
Sometimes this allows removal of conditional or docs

Signed-off-by: Alexander Shopov <ash@kambanaria.org>

### CODE REVIEW

**Comment by alshopov** [comment]
See https://github.com/racket/ChezScheme/pull/9

File: c/new-io.c [hunk 1]
```diff
@@ -383,9 +383,6 @@ ptr S_close_fd(ptr file, IBOOL gzflag) {
  /* refuse to close stdin, stdout, and stderr fds */
   if (!gzflag && fd <= 2) return Strue;
 
- /* file is not locked; do not reference after deactivating thread! */
```
Comment (burgerrg) [inline c/new-io.c]: Please leave the comment here, because it still applies.

File: c/new-io.c [hunk 2]
```diff
@@ -424,8 +421,6 @@ ptr S_bytevector_read(ptr file, ptr bv, iptr start, iptr count, IBOOL gzflag) {
   INT fd = gzflag ? 0 : GET_FD(file);
   glzFile gzfile = gzflag ? GZXFILE_GZFILE(file) : NULL;
 
- /* file is not locked; do not reference after deactivating thread! */
```
Comment (burgerrg) [inline c/new-io.c]: Please leave the comment here, because it still applies.

Comment (alshopov) [reply c/new-io.c]: Comment restored

File: c/new-io.c (hunk 1 above)
Comment (alshopov) [reply c/new-io.c]: Comment restored

**Comment by alshopov** [comment]
> update LOG
Done

**Comment by dybvig** [comment]
That assignment is there to prevent accidental reference.  I'd hate to remove it just because C compilers think it has no purpose.

**Comment by alshopov** [comment]
@dybvig 
> prevent accidental reference

@burgerrg 
> It's easy to overlook the comment

Oh boy, thanx for that! I thought `(ptr)-1` was decrementing 1 from ptr while in fact it is a cast.
I've removed these. What is the reason to use `-1` rather than `NULL`?

**Comment by burgerrg** [comment]
0 is a valid file descriptor (stdin), but -1 isn't. Moreover, (ptr)(-1) is an invalid pointer so that if one tries to dereference it, an exception will be raised.

**Pull request merged**